sdks/python/apache_beam/io/gcsio.py

```diff
@@ -47,7 +47,7 @@
       'Google Cloud Storage I/O not supported for this execution environment '
       '(could not import storage API client).')
 
-DEFAULT_READ_BUFFER_SIZE = 1024 * 1024
+DEFAULT_READ_BUFFER_SIZE = 16 * 1024 * 1024
```

Comment:
For more perspective on why I chose 16 MB, these are the corresponding numbers for 8 MB and 16 MB:

8 MB buffer:
> Read 50 MB in 2.9202067852 seconds (17.1220751398 MB/s)
> Read 100 MB in 4.41199994087 seconds (22.665458146 MB/s)
> Read 200 MB in 8.40019392967 seconds (23.8089741349 MB/s)
> Read 400 MB in 15.3569831848 seconds (26.0467824433 MB/s)

16 MB buffer:
> Read 50 MB in 1.75278306007 seconds (28.5260630017 MB/s)
> Read 100 MB in 2.07984685898 seconds (48.0804630246 MB/s)
> Read 200 MB in 3.68290996552 seconds (54.3048844182 MB/s)
> Read 400 MB in 7.31996202469 seconds (54.6450922356 MB/s)